File: CS/TreeListFilter/FilterTreeList/Forms/CustomFilterForm.cs
using System;
using DevExpress.XtraEditors;
using DevExpress.XtraTreeList;

namespace FilterTreeListControl
{
	public partial class CustomFilterForm : XtraForm
	{
		public CustomFilterForm()
		{
			InitializeComponent();
		}

		private void RemoveConditionFromList(FilterConditionEnum condition)
		{
			int index = cbeFilterConditions.Properties.Items.IndexOf(Enum.GetName(typeof(FilterConditionEnum), condition));
			if ( index != -1 )
				cbeFilterConditions.Properties.Items.RemoveAt(index);
		}

		private void customFilterForm_Load(object sender, EventArgs e)
		{
			cbeFilterConditions.Properties.Items.AddRange(Enum.GetNames(typeof(FilterConditionEnum)));
			RemoveConditionFromList(FilterConditionEnum.Between);
			RemoveConditionFromList(FilterConditionEnum.NotBetween);
			RemoveConditionFromList(FilterConditionEnum.None);
		}

		private void cbeFilterConditions_EditValueChanged(object sender, EventArgs e)
		{
			sbOK.Enabled = ((ComboBoxEdit)sender).Text != "" && teValue.Text != "";
		}

		private void teValue_EditValueChanged(object sender, EventArgs e)
		{
			sbOK.Enabled = ((TextEdit)sender).Text != "" && cbeFilterConditions.Text != "";
		}
	}
}

File: VB/TreeListFilter/FilterTreeList/Forms/CustomFilterForm.vb
Imports System
Imports DevExpress.XtraEditors
Imports DevExpress.XtraTreeList

Namespace FilterTreeListControl
	Partial Public Class CustomFilterForm
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub RemoveConditionFromList(ByVal condition As FilterConditionEnum)
			Dim index As Integer = cbeFilterConditions.Properties.Items.IndexOf(System.Enum.GetName(GetType(FilterConditionEnum), condition))
			If index <> -1 Then
				cbeFilterConditions.Properties.Items.RemoveAt(index)
			End If
		End Sub

		Private Sub customFilterForm_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			cbeFilterConditions.Properties.Items.AddRange(System.Enum.GetNames(GetType(FilterConditionEnum)))
			RemoveConditionFromList(FilterConditionEnum.Between)
			RemoveConditionFromList(FilterConditionEnum.NotBetween)
			RemoveConditionFromList(FilterConditionEnum.None)
		End Sub

		Private Sub cbeFilterConditions_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbeFilterConditions.EditValueChanged
			sbOK.Enabled = (CType(sender, ComboBoxEdit)).Text <> "" AndAlso teValue.Text <> ""
		End Sub

		Private Sub teValue_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles teValue.EditValueChanged
			sbOK.Enabled = (CType(sender, TextEdit)).Text <> "" AndAlso cbeFilterConditions.Text <> ""
		End Sub
	End Class
End Namespace